### ISSUE TITLE
机器学习基础 (Part 1: 基本概念)

### ISSUE THREAD

**Issue by nonocast** [nonocast/me#217]
### 机器学习和深度学习
传统编程是由程序员在程序中内置规则来实现的，机器学习则是通过数据训练得到模型的过程，深度学习是机器学习的一个子集，指利用多层神经网络的一组方法，比如ANN, CNN, RNN。

注: 机器学习不是某种具体的算法，而是很多算法的统称, 这里的算法也称为模型。

### scikit-learn, Tensorflow和PyTorch
可以简单理解scikit-learning是基于NumPy, SciPy和matplotlib开发的一个机器学习的库，有深度学习但比较基础，而Tensorflow和Pytorch则是深度学习的框架。

### 模型和训练

最简单的理解，比如我们在分析一个问题时假设它时一个线性问题，则y=ax+b就是我们选定的模型，然后通过已知的数据对这个模型进行测试 (即推测最佳模型参数的过程)，发现a=2, b=100时最优 (仅在此模型范围内)，这就是说机器通过训练学习到a=2, b=100是最合适的。

常用的机器学习模型:
- Linear Regression
- Logistic Regression
- Decision Tree
- SVM
- Naive Bayes
- kNN
- K-Means
- Random Forest
- Dimensionality Reduction Algorithms
- Gradient Boosting algorithms

在训练过程，损失函数(loss function)的值越小，则模型学习越准确，一般采用梯度下降法。

## 学术和工程

一般来说学术就是写paper然后搞算法，而工程一般把算法模型看成黑盒，通过框架学习参数，然后部署到各个平台进行推理。

参考阅读:
- [深度学习和机器学习有什么区别？ - 知乎](https://www.zhihu.com/question/41268372)

教程:
- [Virgilio-Your new Mentor for Data Science E-Learning. 13k](https://virgili0.github.io/Virgilio/#table-of-contents)
